src/components/Registration/Registration.tsx

```diff
@@ -0,0 +1,113 @@
+import {
+	ChangeEventHandler,
+	FC,
+	FormEventHandler,
+	MouseEventHandler,
+	useState,
+} from 'react';
+import { NavigateFunction, useNavigate } from 'react-router-dom';
+
+import { Input, Button, Label } from '../../common';
+
+import { REGISTRATION_BUTTON_TEXT } from '../../constants';
+
+import { postRegister } from '../../services';
+
+import StyledRegistration from './Registration.styles';
+
+const Registration: FC = () => {
+	const [name, setName] = useState<string>('');
+	const [email, setEmail] = useState<string>('');
+	const [password, setPassword] = useState<string>('');
+	const navigate: NavigateFunction = useNavigate();
+
+	const handleNameChange: ChangeEventHandler<HTMLInputElement> = ({
+		target: { value },
+	}): void => setName(value);
+
+	const handleEmailChange: ChangeEventHandler<HTMLInputElement> = ({
+		target: { value },
+	}): void => setEmail(value);
+
+	const handlePasswordChange: ChangeEventHandler<HTMLInputElement> = ({
+		target: { value },
+	}): void => setPassword(value);
```

Comment:
these handlers are redundant. You can do it in your `onChange` listener. `onChange={ ({ target: { value } }) => setName(value) }`